Standards/scs-0112-v1-volume-type-standard.md

```diff
@@ -0,0 +1,227 @@
+| properties         | encrypted='true'                                                                                                                                             |
+| qos_specs_id       | None                                                                                                                                                         |
++--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+We want to find a way to also use the internal extra_spec for replication, when the replication is automatically done by the backend. If this is not possible, we would like to introduce another property, which has to be set by the admin, when setting the volume type. Only after that we will have the possibility to automatically check for a volume type with replication.
+
+## DEFAULT volume types
+
+There is always a default volume type defined in an OpenStack deployment. The SCS does not have any requirements about this volume type at this moment, instead deployers are free to choose what fits best in their environment. Conversely, a cloud user can not expect any specific behavior or properties from default volume types.
+
+The parameters of volume types described in this standard do not have to be applied to the chosen default volume type. And the SCS will not make any assumptions about parameters being present in the default volume type.
+
+## REQUIRED volume types
+
+Currently the SCS will not require volume types with certain specification. This will change in the future.
+
+## RECOMMENDED volume types
+
+The SCS recommends to have one or more volume types, that satisfy the need for encrpytion and replication. These are all allowed possibilites for those recommended volume types:
+
+| Volume Type Nr | Type Name | Volume Type Description |
+| ---- | ---- | ---- |
+| 1 | `__DEFAULT__` | `[encrypted][replicated] Volumes will be encrypted and three times replicated` |
+| 2 | `LUKS` | `[encrypted] This volume type creates LUKS encryped volumes` |
+| 3 | `replicated` | `[replicated] Volumes will be three times replicated` |
+
+The standardized phrases are `[encrypted]` and `[replicated]`. They MUST appear at the beginning of the description of the volume type and they MUST be ordered alphabetically. Having only volume type number 1 would be enough to fulfill all recommendations. Having only volume types 2 and 3 together would also satisfy all recommendations
+
+### Encryption
+
+Encryption for volumes is an option which has to be configured within the volume type. As an admin it is possible to set encryption-provider, key size, cipher and control location. And for admins it is also currently possible to see these configurations in a volume type with both list and show commands.
```

Comment:
As this standard RECOMMENDs encrypted volumes, is MUST make recommendations at least for cipher and key-size as well. Cloud customers will buy a pig a poke otherwise.

Reply:
well I just want to point out: we did debate this during one of the calls - I don't remember when exactly - and I also voiced concerns regarding this.

I can't remember the details (like "why"), but it was agreed upon to explicitly _not_ encode details about the used encryption scheme into the standard.